flow/project.py

```diff
@@ -1469,10 +1466,10 @@ def __call__(self, func, name=None):
                 self._parent_class._GROUPS.append(
                     FlowGroupEntry(name=name, project=self._parent_class, options="")
                 )
-                if hasattr(func, "_flow_groups"):
-                    func._flow_groups.add(name)
-                else:
-                    func._flow_groups = {name}
+                if not hasattr(func, "_flow_groups"):
+                    func._flow_groups = {}
+                if self._parent_class not in func._flow_groups:
+                    func._flow_groups[self._parent_class] = {name}
```

Comment:
This check should be unnecessary as if this is true then this function is already in another group with the same project which shouldn't happen.
```suggestion
                func._flow_groups[self._parent_class] = {name}
```